### PR TITLE
STOEGE-19591 _helpers.tpl & app-cm.yaml refactoring

### PR DESCRIPTION
**Pull request by SuslikDaRete**

### CODE REVIEW

File: chart/templates/app-cm.yaml [hunk 1]
```diff
@@ -64,19 +64,18 @@ data:
     algorithm = "ES256"
     key = "data/keys/svc.private_key.p8.der"
 
-    {{- $svc_audience := list .audience | compact | join "." }}
+    {{- $audience := .audience }}
```
Comment (manifest) [inline chart/templates/app-cm.yaml:67]: не понятно какая именно audience здесь

File: chart/templates/_helpers.tpl [hunk 2]
```diff
@@ -51,23 +51,35 @@ app.kubernetes.io/instance: {{ .Release.Name }}
...
+{{- define "conference.serviceAudience" -}}
+{{- $audience := . -}}
+{{- list "svc" $audience | join "." -}}
+{{- end -}}
+
+{{/*
+User Audience
+*/}}
+{{- define "conference.userAudience" -}}
+{{- $audience := . -}}
```
Comment (manifest) [inline chart/templates/_helpers.tpl:65]: Здесь `audience` это `tenant`. А на выходе получается tenant user audience.

**Review by manifest — APPROVED** [review]

File: chart/templates/app-cm.yaml (hunk 1 above)
Comment (SuslikDaRete) [reply chart/templates/app-cm.yaml:67]: это audience самого микросервиса

File: chart/templates/_helpers.tpl (hunk 2 above)
Comment (SuslikDaRete) [reply chart/templates/_helpers.tpl]: обновил

**Review by manifest — APPROVED** [review]

File: chart/templates/app-cm.yaml [hunk 3]
```diff
@@ -106,43 +105,43 @@ data:
...
+    uri = {{ default $defaultAuthzUrl .uri | quote }}
     algorithm = {{ default "ES256" .algorithm | quote }}
     key = {{ default "data/keys/svc.private_key.p8.der" .key | quote }}
     user_agent = {{ default $id .user_agent | quote }}
     max_retries = {{ default 3 .max_retries }}
     trusted = [
-      {{- range $account_label := .trusted }}
-        {{ (list $account_label $.Values.app.svc.audience) | compact | join "." | quote }},
+      {{- range $accountLabel := .trusted }}
+        {{ (list $accountLabel $.Values.app.svc.audience) | compact | join "." | quote }},
```
Comment (manifest) [inline chart/templates/app-cm.yaml:132]: This entire `trusted` block seems to be a duplicate of the following one.

**Review by manifest — CHANGES_REQUESTED** [review]

File: chart/templates/_helpers.tpl [hunk 4]
```diff
@@ -51,23 +51,35 @@ app.kubernetes.io/instance: {{ .Release.Name }}
...
-{{- define "conference.audienceEnvironment" -}}
-{{- $v := regexReplaceAll "(s)(\\d\\d)" (include "conference.shortNamespace" .) "staging${2}" }}
-{{- $v := regexReplaceAll "(t)(\\d\\d)" $v "testing${2}" }}
-{{- $v }}
+{{- define "conference.tenantObjectAudience" -}}
+{{- $vars := index . 0 -}}
+{{- $tenant := index . 1 -}}
+{{- $env := regexSplit "-" $vars.Release.Namespace -1 | first -}}
+{{- $devEnv := ""}}
+{{- if eq $env "p" }}
```
Comment (manifest) [inline chart/templates/_helpers.tpl:77]: Shouldn't we return `$tenant` in this case?

File: chart/templates/app-cm.yaml [hunk 5]
```diff
@@ -106,43 +105,43 @@ data:
...
       {{- end }}
     ]
     {{- end }}
     {{- if eq "local" .type }}
     trusted = [
-      {{- range $account_label := .trusted }}
-        {{ $ns_audience | list $account_label | join "." | quote }},
+      {{- range $accountLabel := .trusted }}
+        {{ list $accountLabel $tenantObjectAudience | join "." | quote }},
```
[the remark this answers is not in the document]
Comment (SuslikDaRete) [reply chart/templates/app-cm.yaml]: Yeap, you're right
Fixed

File: chart/templates/app-cm.yaml (hunk 3 above)
Comment (SuslikDaRete) [reply chart/templates/app-cm.yaml:125]: Nope. It's in different `if` conditions

File: chart/templates/_helpers.tpl (hunk 4 above)
Comment (SuslikDaRete) [reply chart/templates/_helpers.tpl]: Not here. Rewrote the condition to make it clearer

File: chart/templates/app-cm.yaml [hunk 6]
```diff
@@ -98,51 +93,48 @@ data:
...
     ## {{ .audience }}
     ##
-    {{- $svc_audience := list "svc" .audience | compact | join "." }}
-    {{- $usr_audience := list "usr" .audience | compact | join "." }}
-    {{- with .authn }}
-    [authn.{{ list "iam" $svc_audience | join "." | quote }}]
-    audience = [{{ $svc_audience | quote }}, {{ $usr_audience | quote }}]
+    {{- $tenantServiceAudience := include "conference.tenantServiceAudience" .audience }}
+    {{- $tenantUserAudience := include "conference.tenantUserAudience" .audience }}
+    {{- $tenantObjectAudience := include "conference.tenantObjectAudience" (list $ .audience) }}
```
Comment (manifest) [inline chart/templates/app-cm.yaml:105]: Why do we have `(list $ .audience)` instead of just `.audience` here?

**Review by manifest — APPROVED** [review]

**Review by manifest — APPROVED** [review]

**Pull request merged**